local/network.go

```diff
@@ -714,6 +714,55 @@ func (ln *localNetwork) removeNode(ctx context.Context, nodeName string) error {
 	return nil
 }
 
+// Restart [nodeName] using the same config, optionally changing [binaryPath],
+// [buildDir], [whitelistedSubnets], [dbDir]
+func (ln *localNetwork) RestartNode(
+	ctx context.Context,
+	nodeName string,
+	binaryPath string,
+	whitelistedSubnets string,
+	dbDir string,
+) error {
+	ln.lock.Lock()
+	defer ln.lock.Unlock()
+
+	node, ok := ln.nodes[nodeName]
+	if !ok {
+		return fmt.Errorf("node %q not found", nodeName)
+	}
+
+	nodeConfig := node.GetConfig()
+
+	if binaryPath != "" {
+		nodeConfig.BinaryPath = binaryPath
+		nodeConfig.Flags[config.BuildDirKey] = filepath.Dir(binaryPath)
+	}
+
+	if whitelistedSubnets != "" {
+		nodeConfig.Flags[config.WhitelistedSubnetsKey] = whitelistedSubnets
+	}
+
+	if dbDir != "" {
```

Comment:
Why would restarting a node actually allow to change the dbDir? Isn't it the same node?

Reply:
double checked. currently there is no need to accept dbdir, rootdir as param to restart.